### PR TITLE
IHostedService StopAsync cancellationToken updates

### LINKED ISSUE #9144: What happens with IHostedService.StopAsync when the cancellationToken is set cancelled?
# Issue description

The documentation for IHostedService.StopAsync doesn't explain what the parameter is used for and what happens when it is used. Will the host just ignore these services and quit anyway? Or is this just an indication that we should hurry up? I don't know what to do with this parameter in my IHostedService implementation.

# Software versions

Check the .NET target framework(s) being used, and include the version number(s).

- [x] .NET Core 2.1
- [ ] .NET Framework
- [ ] .NET Standard

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #9144 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-2.1&branch=pr-en-us-9555#ihostedservice-interface)

cc: @ygoe

### CODE REVIEW

**Comment by guardrex** [comment]
@jkotalik Here's another really quick one ... it's just Line 40. Just trying to flesh out `StopAsync` and the cancellation token a bit more.

File: aspnetcore/fundamentals/host/hosted-services.md [hunk 1]
```diff
@@ -33,9 +33,13 @@ Reference the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapacka
...
+* [StartAsync(CancellationToken)](xref:Microsoft.Extensions.Hosting.IHostedService.StartAsync*) &ndash; `StartAsync` contains the logic to start the background task. When using the [Web Host](xref:fundamentals/host/web-host), `StartAsync` is called after the server has started and [IApplicationLifetime.ApplicationStarted](xref:Microsoft.AspNetCore.Hosting.IApplicationLifetime.ApplicationStarted*) is triggered. When using the [Generic Host](xref:fundamentals/host/generic-host), `StartAsync` is called before `ApplicationStarted` is triggered.
 
-* [StopAsync(CancellationToken)](xref:Microsoft.Extensions.Hosting.IHostedService.StopAsync*) - Triggered when the host is performing a graceful shutdown. `StopAsync` contains the logic to end the background task and dispose of any unmanaged resources. If the app shuts down unexpectedly (for example, the app's process fails), `StopAsync` might not be called.
+* [StopAsync(CancellationToken)](xref:Microsoft.Extensions.Hosting.IHostedService.StopAsync*) &ndash; Triggered when the host is performing a graceful shutdown. `StopAsync` contains the logic to end the background task and dispose of any unmanaged resources.
```
Comment (Tratcher) [inline aspnetcore/fundamentals/host/hosted-services.md]: ~~and dispose of any unmanaged resources~~ Implement IDisposable and finalizers for that.

File: aspnetcore/fundamentals/host/hosted-services.md [hunk 2]
```diff
@@ -33,9 +33,18 @@ Reference the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapacka
...
 
-* [StopAsync(CancellationToken)](xref:Microsoft.Extensions.Hosting.IHostedService.StopAsync*) - Triggered when the host is performing a graceful shutdown. `StopAsync` contains the logic to end the background task and dispose of any unmanaged resources. If the app shuts down unexpectedly (for example, the app's process fails), `StopAsync` might not be called.
+* [StopAsync(CancellationToken)](xref:Microsoft.Extensions.Hosting.IHostedService.StopAsync*) &ndash; Triggered when the host is performing a graceful shutdown. `StopAsync` contains the logic to end the background task. Implement <xref:System.IDisposable> and [finalizers (destructors)](/dotnet/csharp/programming-guide/classes-and-structs/destructors) to dispose of any unmanaged resources.
```
Comment (Tratcher) [inline aspnetcore/fundamentals/host/hosted-services.md]: You've duplicated the IDisposable comments here and two paragraphs down.

Comment (guardrex) [reply aspnetcore/fundamentals/host/hosted-services.md]: Thanks ... I look line-by-line tooooooooo often on reviews. It's *another one* of my faults around here. :cry:

**Pull request merged**